main/autogen/data/version.h

```diff
@@ -0,0 +1,14 @@
+#ifndef AUTOGEN_VERSION_H
+#define AUTOGEN_VERSION_H
+
+namespace sorbet::autogen {
+
+class AutogenVersion {
+public:
+    constexpr static int MIN_VERSION = 2;
+    constexpr static int MAX_VERSION = 3;
+};
+
+} // namespace sorbet::autogen
```

Comment:
Pull out `MIN_VERSION` / `MAX_VERSION` into a different wrapper class so that `autogen.cc` does not have to depend on `msgpack.cc`